cfpb/consumerfinance.gov
# Switch to ruff for all our Python linting and formatting

Proposal: Switch to ruff for all our Python linting and formatting needs.

This replaces Black and isort with ruff, and enables ruff's support for pyupgrade and flake8-simplify checks, and then makes all the necessary formatting changes to comply with those changes.

Anecdotally, pre-commit's hooks are a lot faster locally than they were with black and isort, as is `tox -e lint`

## How to test this PR

1. Test pre-commit (with changes that should break and without): `pre-commit run`
1. Test tox listing (with changes that should break and without): `tox -e lint`
1. Test `ruff format` and `ruff check` with changes that should break and without

## How to review this PR

I tried to split the commits up so that all of the configuration is in the first PR; commits that are automatic changes by `ruff format`, `ruff check --fix`, and `ruff check --fix --unsafe-fixes` include the command used in the extended commit message, and changes I've manually made are described in manual commits. 

For the most part though, the changes that have to be made to formatting to pass linting checks are necessary. Some things (like when to use `noqa` and when not to) are within our control.


## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing



#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
